packages/ihe-gateway-sdk/src/client/ihe-gateway.ts

```diff
@@ -0,0 +1,44 @@
+import { Patient } from "@medplum/fhirtypes";
+import { v4 as uuidv4 } from "uuid";
+import { principalCareProviderIdsSchema } from "../models/shared";
+import { XCPDGateways, XCPDRequest, xcpdGatewaysSchema } from "../models/xcpd-request";
+
+export class IHEGateway {
+  /**
+   * An XCPD request interface
+   * @param patient The patient data in FHIR R4 format.
+   * @param cxId The customer ID.
+   * @param xcpdGateways The OIDs and XCPD ITI-55 urls of each organization to make a request to.
+   * @param principalCareProviderIds The list of NPIs of the practitioners associated with the patient.
+   * @param requestId Optional. Unique id for the request. If not provided, one will be created.
+   * @returns an XCPD request to be used with an IHE Gateway.
+   */
+  createXCPDRequest({
+    patient,
+    cxId,
+    xcpdGateways,
+    principalCareProviderIds,
+    requestId,
+  }: {
+    patient: Patient;
+    cxId: string;
+    xcpdGateways: XCPDGateways;
+    principalCareProviderIds?: string[];
+    requestId?: string;
+  }): XCPDRequest {
+    const xcpdRequest: XCPDRequest = {
+      id: requestId ?? uuidv4(),
+      cxId,
+      xcpdGateways: xcpdGatewaysSchema.parse(xcpdGateways),
+      timestamp: new Date().toISOString(),
+      patientResource: patient,
+    };
+
+    if (principalCareProviderIds) {
+      xcpdRequest.principalCareProviderIds =
+        principalCareProviderIdsSchema.parse(principalCareProviderIds);
+    }
```

Comment:
`if (principalCareProviderIds)` ?

we should name using NPIs mandatory, no? would suggest checking the array to make sure we validate all NPIs and that at least 1 is valid

Reply:
I have a schema parsing that validates all NPIs - is that not enough?

Reply:
why don't we use an object made from that schema as the input to this function then?

the caller could easily shoot themselves in the foot by passing random strings in